corehq/apps/userreports/models.py

```diff
@@ -153,6 +154,9 @@ def data_source_id(self):
     def get_columns(self):
         raise NotImplementedError()
 
+    def pk_columns(self):
```

Comment:
This should probably have an `@property` decorator. 
```python
    >>> class Foo(object):
    ...     def bar(self):
    ...         raise NotImplementedError()
    ... 
    >>> foo = Foo()
    >>> foo.bar = 'hello'
    #crickets
```

Reply:
good catch